### PR TITLE
Refactor Sleep Class to DRY up methods

### LINKED ISSUE #23: Refactor Sleep Class
There are several functions that can be combined into one dynamic function and DRY up the Sleep Class.

### PR DESCRIPTION
**Pull request by Dustin-Mikusko**
#### What's this PR do?
This PR DRYs up the Sleep Class by refactoring the methods to be more dynamic, thus using less overall code. It also modifies the Sleep-test file to coincide with the changes and pass the method tests. 
#### Where should the reviewer start?
Reviewer should start in the `Sleep.js` file to see changes. The `Sleep-test.js` file also has test updates to match the changes in the `Sleep.js` file. 
#### How should this be manually tested?
Pull down this branch, run both `npm run test` and `npm run start` to make sure your local machine passes the tests, and the localhost shows the `script.js` updates.
#### What are the relevant tickets?
This PR closes #23 
#### Screenshots (if appropriate)
UNADJUSTEDNONRAW_thumb_14cb
UNADJUSTEDNONRAW_thumb_14cc
UNADJUSTEDNONRAW_thumb_14cd
UNADJUSTEDNONRAW_mini_14ce
Image 10-21-19 at 6 05 PM
Image 10-21-19 at 6 06 PM

### CODE REVIEW

File: src/Sleep.js [hunk 1]
```diff
@@ -17,42 +17,24 @@ class Sleep {
...
 
-  returnAvgSleepHours() {
+  returnAvgSleepInfo(sleepInfo) {
     let specificUser = this.findUser();
-    return Number((specificUser.reduce((totalHours, day) => {
-      totalHours += day.hoursSlept;
-      return totalHours;
+    return Number((specificUser.reduce((acc, day) => {
+      acc += day[sleepInfo];
+      return acc;
```
Comment (adamsjr8576) [inline src/Sleep.js]: This all looks good Dustin - this is a nice use of bracket notation coupled with using a parameter - it's crazy how much less code this creates! good refactoring!

File: src/Sleep.js [hunk 2]
```diff
@@ -17,42 +17,24 @@ class Sleep {
...
-  returnWeekOfSleepHours(week) {
+  returnSleepInfo(date, sleepInfo) {
     let specificUser = this.findUser();
-    return this.returnWeekOfData(week, specificUser).map(day => day.hoursSlept);
+    return specificUser.find(day => day.date === date)[sleepInfo];
   }
 
-  returnWeekOfSleepQuality(week) {
+  returnWeekOfSleepInfo(week, sleepInfo) {
     let specificUser = this.findUser();
```
Comment (adamsjr8576) [inline src/Sleep.js]: I'm about to push up my work on the inherit stuff and I was able to get rid of the specificUser = this.findUser() on pretty much every function using inherit so these functions will be real clean once we combine these!

File: test/Sleep-test.js [hunk 3]
```diff
@@ -47,27 +47,27 @@ describe('Sleep', () => {
...
   });
 
   it('should return hours slept each day for week for a specific user', () => {
-    expect(fullSleep.returnWeekOfSleepHours(2)).to.eql([7.3, 5.1, 8.6, 10.5, 9.1, 6.5, 6.8]);
+    expect(fullSleep.returnWeekOfSleepInfo(2, 'hoursSlept')).to.eql([7.3, 5.1, 8.6, 10.5, 9.1, 6.5, 6.8]);
   });
 
   it('should return hours slept each day for week for a specific user', () => {
-    expect(fullSleep.returnWeekOfSleepQuality(2)).to.eql([4.8, 4.7, 3.7, 1.8, 1.5, 4.2, 2]);
+    expect(fullSleep.returnWeekOfSleepInfo(2, 'sleepQuality')).to.eql([4.8, 4.7, 3.7, 1.8, 1.5, 4.2, 2]);
```
Comment (adamsjr8576) [inline test/Sleep-test.js]: Nice work updating the tests and the scripts file - this all looks good to go!

**Comment by Jessmaxim303** [comment]
Great job removing specificUser. Makes everything a lot easier to deal with.

**Pull request merged**